dleq/go_dleq.go

```diff
@@ -0,0 +1,76 @@
+package dleq
+
+import (
+	csecp256k1 "github.com/athanorlabs/atomic-swap/crypto/secp256k1"
+
+	dleq "github.com/noot/go-dleq"
+	"github.com/noot/go-dleq/ed25519"
+	"github.com/noot/go-dleq/secp256k1"
+
+	dsecp256k1 "github.com/decred/dcrd/dcrec/secp256k1/v4"
+)
+
+// DefaultDLEq is the default DLEq prover.
+type DefaultDLEq = GoDLEq
+
+// GoDLEq is a wrapper around the go-dleq library prover and verifier.
+type GoDLEq struct{}
+
+var (
+	curveA = secp256k1.NewCurve()
+	curveB = ed25519.NewCurve()
+)
+
+// Prove generates a secret and a corresponding proof that it has a value
+// on the secp256k1 and ed25519 curves.
+func (d *GoDLEq) Prove() (*Proof, error) {
+	x, err := dleq.GenerateSecretForCurves(curveA, curveB)
+	if err != nil {
+		panic(err)
```

Comment:
Can I assume this is panicking because the called function should never fail? If that's the reason, can we add a comment?

Reply:
i forgot to change it, updated